src/annbatch/io.py

```diff
@@ -123,25 +120,54 @@ def _create_chunks_for_shuffling(adata: ad.AnnData, shuffle_n_obs_per_dataset: i
     return idxs
 
 
+def _persist_adata_in_memory(adata: ad.AnnData) -> ad.AnnData:
+    adata.X = adata.X.persist()
+
+    if isinstance(adata.obs, Dataset2D):
+        adata.obs = adata.obs.to_memory()
+    if isinstance(adata.var, Dataset2D):
+        adata.var = adata.var.to_memory()
+    if adata.raw is not None:
+        adata_raw = adata.raw.to_adata()
+        if isinstance(adata_raw.X, DaskArray):
+            adata_raw.X = adata_raw.X.persist()
+        if isinstance(adata_raw.var, Dataset2D):
+            adata_raw.var = adata_raw.var.to_memory()
+        if isinstance(adata_raw.obs, Dataset2D):
+            adata_raw.obs = adata_raw.obs.to_memory()
+        del adata.raw
+        adata.raw = adata_raw
+
+    for k, elem in adata.obsm.items():
+        if isinstance(elem, DaskArray):
+            adata.obsm[k] = elem.persist()
```

Comment:
```suggestion
        # TODO: handle `Dataset2D` in `obsm` and `varm` that are 
        if isinstance(elem, DaskArray):
            adata.obsm[k] = elem.persist()
```

technically you can do this, but don't want to block on it